src/config.rs

```diff
@@ -713,13 +713,14 @@ impl<'a> Cfg<'a> {
         &self,
         name: Option<ResolvableToolchainName>,
     ) -> Result<Toolchain<'_>> {
-        Ok(match name {
+        let toolchain = match name {
```

Comment:
Would this be better with `map().transpose()`?

Reply:
@djc I remember I was discouraged from using combinators when there are nested `Result`s and `Option`s? https://github.com/rust-lang/rustup/pull/3948#discussion_r1677450531

Reply:
@djc I plan to at least include another refactoring that brings two functions together... I guess we can leave this as-is until then.

> This PR also makes it possible to merge find_or_install_active_toolchain() and find_active_toolchain() by reducing the number of call sites of the former.

Reply:
Oh I see, your idea comes from `resolve_local_toolchain()`. I feel okay about this, it's just that the current version makes it more clear about how the refactoring is done.

Reply:
@djc I've made https://github.com/rust-lang/rustup/pull/3986 for this.